### ISSUE TITLE
bionics official release

### ISSUE THREAD

**Issue by mrfricks** [puppylinux-woof-CE/woof-CE#1354]
@puppylinux-woof-CE/members /peebee
so two bionic based puppies are ready for release, upupbionic and bionicpup64.. i think that they should be released together as a 32bit & 64bit.

i think that releasing separately would be confusing to users (new users, not just puppy forum peeps) the spiel should big up their differences, the 32bit designed for older comps and the 64bit designed for modern comps by two different devs.  i've always liked this difference and think that it's a good selling point that they are so different.

so now we have to broach the problem that ..... upupbionic has a completely out of order version system to bionicpup64 and puppy in general. i agree that it might make better sense ultimately, but it would mean a complete change of versioning system for puppy going forward, and make past versioning defunct.

so everyone @peebee what shall we do ... as we need to release.

**Comment by mrfricks** [comment]
@peebee i'm happy for upupbionic to be the screenshot on distrowatch and the main link to the iso... but i feel really strongly that they need to be released together

**Comment by mavrothal** [comment]
I expressed my views on that few days back but I'm afraid is not going to be easily done. ie is not just changing distro specs. 
The repos must be renamed making existing upupbb unusable or duplicated, unless ibiblio supports soft links (does it?)
It also needs for the ISOs at least to be located in ibiblio as the release needs a direct link. 
Currently it redirects to sourceforge and I'm not sure if this is OK with DW to say the lest. (It also leaves sourceforge as the ONLY source of the files. No ibiblio and no mirrors but that's a minor issue for the moment).
So I would think that for a common/dual  release to happen at this point, upupbb and bionicpup32 (or whatever the final name might be) should exist in parallel. 
It is a hassle and certainly not fun at all but I think alternatives (not releasing, releasing one of the two, try to release independently, release another "competitive" 32 bit version 😮 ) are not doing justice to persons involved or the community and the distro.
Hopefully @peabee is willing to go through the hassle. 🤞

**Comment by peabee** [comment]
I am thinking about what would need to be done and what I want to do.....
There is a lot of work involved in changing UPupBB to BionicPup32..... @mavrothal is correct in the matters mentioned (ibiblio will do symlinks - there is actually no UPupBB repo - it is just a link to the Xenial32 repo).
SF has much better mirroring than ibiblio but ibiblio does mirror the existing arrangements so this is not an issue
Besides the work needed to change the name, the build should change from its current 'master' branch to the current 'testing' branch of woof-ce
This would mean I think that the new version would need to have a pre-release for testing as it would be considerably changed from what has been tested so far.
And I worry that my build includes adrv, fdrv and a zdrv with no firmware and has entirely different update mechanisms to any other pup....
Hmmmmmmmmmmm...........!!!!!!!!!!!!!!

**Comment by peabee** [comment]
OK - I think the line of least resistance is to:
1. rebuild UPupBB using 'testing' branch of woof-ce
2. keep all internal components of the .iso with their current UPupBB names and versioning
but
3. rename a copy of the current .iso to: bionicpup32-8.0-uefi.iso
4. make 2 subdirectories in http://distro.ibiblio.org/puppylinux/puppy-bionic - bionicpup64 and bionicpup32
5. move current bionicpup64-8.0 elements into the bionicpup64 subdirectory
6. put equivalent bionicpup32-8.0 elements into the bionicpup32 subdirectory (the devx will also be renamed to match the .iso but will therefore be different to the .iso components)

Anybody see any problems with this 'strategy'? The default wallpaper could be changed I guess to say: "BionicPup32 (UPup Bionic Beaver)"

**Comment by peabee** [comment]
Then there's the kernel - 4.9.96 is getting 'old' - now 4.9.162 - but changing would mean updates to kernel related add-ons.......oh dear!

**Comment by peabee** [comment]
BUILD_FROM_WOOF='testing;0c9e0a1c;2019-03-13 09:51:50 +0000'
eb382462b53d39c23c2fa76fa376eb54  bionicpup32-8.0-uefi.iso

https://my.pcloud.com/publink/show?code=XZ08vq7Z6lrjOWjYePJ6CFTIULQpSYRdCDv7

![Screenshot](https://user-images.githubusercontent.com/5926865/54299965-dfefad80-45b3-11e9-8a79-4ebccc10bf6a.png)

**Comment by mavrothal** [comment]
On a VM ISO (md5sum OK) fails with ```Failed to load ldlinux.c32```

**Comment by mavrothal** [comment]
OK in manual frugal.
ISO boot fails in both EFI and BIOS boot

**Comment by peabee** [comment]
These sorts of things were why I hadn't changed from Master to Testing up to now....
I'm pretty sure I know whats wrong and will do another build tomorrow sometime.
Thanks for testing. I think more testing by the 'community' will be needed before we can be confident to release. I'll pull the test iso.

**Comment by mavrothal** [comment]
Played with it some more. Did not see any there showstoppers.
PPM appears OK. Browser download, icons, utilities, apps, theming etc ok. Low RAM and CPU usage. Pretty solid overall. 
Kernel is old but that could change easily (even if proprietary drivers come a couple of days latter)

**Comment by mrfricks** [comment]
hi peebee,
I've now moved bionicpup64 stuff to its own directory in puppy-bionic.
I don't think you need to do a complete new build with a different woofce branch. This would probably just cause you lots of headaches!
renaming the iso and the new wallpaper is all that's probably needed, and you should check that this isn't going to cause problems with your update mechanism.

**Comment by peabee** [comment]
daf9c22c7b82f713b69c9ee85b49dc6c  bionicpup32-8.0-uefi.iso

https://sourceforge.net/projects/zestypup/files/Upup-Bionic/test/bionicpup32-8.0-uefi.iso/download

**Comment by wdlkmpx** [comment]
Before bionicpup is officially released i think there's something that should be done:

Move Packages-puppy-noarch-official to woof-CE-noarch

All the team members will be able to edit woof-CE-noarch/Packages-puppy-noarch-official and  to add packages to woof-CE-noarch/pet_packages-noarch/ .. via the github web interface... no need to learn to use the git cli app.

That way anyone wanting to update or add noarch pet packages can easily request a change via a pull request.

This solves a major issue, but all the DISTRO_PET_REPOS files must be edited again

It would take a few days. Does anybody agree?

**Comment by peabee** [comment]
Would be OK with me - there are 248 entries in noarch so moving them is quite a task unless it can be automated....
There would be the probably minor problem that old builds out in the wild would still point to ibiblio and wouldn't "see" any changes made to pets updated on github....but they would be using the github Packages-puppy-noarch-official and if they updated databases in PPM there would be an inconsistency?

**Comment by mrfricks** [comment]
i'm happy with that.

**Comment by wdlkmpx** [comment]
I was finishing importing the noarch repo to woof-CE-noarch when i
noticed a few empty files.

These files were not found:

gnu-unifont-5.1-noarch.pet
os-prober-1.42-noarch.pet
type1-fonts-0.1-noarch.pet

The langpacks are big and no longer apply, basically all the woofce
scripts have changed too much and will continue to change, many
scripts were deleted.

some langpacks have edited woof scripts. So i won't upload the langpacks

The right thing to do is create new langpacks and the extra
translations must come from the build system.

The build system should optionally create _DOC.sfs and _NLS.sfs

**Comment by wdlkmpx** [comment]
Aye. Hurr.

Ok, that's already done, although it can be completely undone.

With a PET_REPOS pointing to the woof-CE-noarch pet repo, updated packages can be retrieved in the future, including updated packages that currently are only in rootfs-packages/

It really depends on you. Revert to the ibiblio noarch repo or not. To be or not to be.

My advice is that you release bionicpup 8.1 and delete the 8.0 iso saying you found another savefile massacre bug or something. Then notify the watchdistro staff or something,, i never visit that page.

I hope this message is clear.

I've been typing a lot trying to explain things, in commits, issues, pull requests, so i'm kind of tired, for me writing in english is not that easy, it's not part of my daily activities, only when it comes to software and trying to explain things to people, it's just has a bit harder than writing a bash script.
